### ISSUE TITLE
oehostfs,fprintf calls after oe_terminate_enclave(), called by internal enclave functions crash the application with segsev

### ISSUE THREAD

**Issue by RRathna** [openenclave/openenclave#4349]
This scenario was met when enabling logging callback for enclave logs (#4323).

When the enclave callback function set calls 'fprintf', the program crashes after oe_terminate_enclave() call is made. This is probably because the enclave log (which is generated as part of oe_terminate_enclave() call), is trying to use memory address that has already been deallocated to print the log into (even in case of stdout, stderr etc) and throws a segmentation fault.

This needs to be further investigated, to find out if this is solvable.

**Comment by anakrish** [comment]
Good catch. Beyond a point during enclave termination, it will not be possible to make ocalls since the allocator has been destructed. Beyond this point, logging won't work.

**Comment by radhikaj** [comment]
@anakrish  - Can we mark this as closed as this is by design?

**Comment by anakrish** [comment]
> @anakrish - So this behavior is by design? We should document this.

I don't think it is `by design`. It wasn't carefully thought out.

Ideally, we wan't to disable OCALLs, logging etc after a certain point during termination.
@mingweishih Any thoughts?

**Comment by mingweishih** [comment]
> > @anakrish - So this behavior is by design? We should document this.
> 
> I don't think it is `by design`. It wasn't carefully thought out.
> 
> Ideally, we wan't to disable OCALLs, logging etc after a certain point during termination. @mingweishih Any thoughts?

Yeah, I think we just need figure out what's the point is, and provide better user experience (e.g., putting a log when we hit that point). Ideally, any logging after that point we should also handle it with a better manner (e.g., making them no-op instead of crashing).

**Comment by anakrish** [comment]
Can you try out the fix in: https://github.com/openenclave/openenclave/pull/4355
